define( "Araport/View/Dialog/UserGuide", [
            'dojo/_base/declare',
            'JBrowse/View/InfoDialog'
        ],
        function(
            declare,
            InfoDialog
        ) {
return declare( InfoDialog, {

    title: "Detailed User Guide",

    constructor: function(args) {
        this.browser = args.browser;
        this.defaultContent = this._makeDefaultContent();

        if( ! args.content && ! args.href ) {
            // make a div containing our help text
            this.content = this.defaultContent;
        }
    },

    _makeDefaultContent: function() {
        return    ''
                + '<div class="help_dialog">'
                + '<div class="main" style="float: left; width: 79%;">'

                + '<dl>'
                + '<h1>JBrowse User Guide</h1>'
                + '<dd><ul>'
	        + '    <li><a href="#setup">Setup and Configuration Documentation</a></li>'
	        + '    <li><a href="#navigation">Navigation</a></li>'
	        + '    <li><a href="#search">Search</a></li>'
	        + '    <li><a href="#data">Data</a></li>'
		+ '    <li><a href="#tracklists">Track Lists</a></li>'
	        + '    <li><a href="#newfeatures">New Features</a></li>'
	        + '    <li><a href="#FutureDevelopments">Future Developments</a></li>'
	        + '    <li><a target="_blank" href="https://www.araport.org/help/faq#t348n239">FAQ</a></li>'
	        + '    <li><a target="_blank" href="https://www.araport.org/jbrowse/user-guide">View in Araport</a></li>'
                + '</ul></dd>'
                + '<h2 id="setup">Setup and Configuration Documentation</h2>'
                + '<dd><ul>'
	        + '    <li><a target="_blank" href="docs/tutorial/">Quick-start tutorial</a></li>'
		+ '    <li><a target="_blank" href="http://gmod.org/wiki/JBrowse">JBrowse Configuration Guide</a></li>'
	        + '    <li><a target="_blank" href="docs/config.html"><tt>biodb-to-json.pl</tt> reference</a></li>'
		+ '    <li><a target="_blank" href="docs/featureglyphs.html">HTMLFeatures class reference</a></li>'
                + '</ul></dd>'
                + '<h2 id="navigation">Navigation</h2>'
                + '<dd><ul>'
	        + '    <li>JBrowse has a simple double pane interface, with the list of tracks on the left side and the display of features on the right. The navigation and search panel are located right below the menu bar.</li>'
                + '</ul></dd>'
	        + '<img src="plugins/Araport/img/jbrowse_intro.jpg" height="420px">'
                + '<dd><ul>'
	        + '<dt>Moving</dt>'
                + '    <li>You can move across the genome by clicking and dragging your mouse inside the track window or by using the navigation tools <img src="plugins/Araport/img/nav_arrows.jpg" height="15px">  or by pressing the left and right arrow keys on your keyboard.</li>'
                + '    <li>Center the view at a point by clicking on either the track scale bar or overview bar, or by shift-clicking in the track area.</li>'
                + '</ul></dd>'
                + '<dd><ul>'
	        + '<dt>Zooming</dt>'
                + '    <li>Zoom in and out by clicking zoom buttons <img src="plugins/Araport/img/nav_zoom.jpg" height="20px"> in the navigation bar or by pressing the up and down arrow keys + shift. Select a region and zoom to it ("rubber-band" zoom) by clicking and dragging in the overview or track scale bar, or shift-clicking and dragging in the track area. Some tracks allow you to zoom into a feature as one of its right-click(CTRL+CLICK on MAC) drop down menu options. </li>'
                + '    </ul>'
	        + '<img src="plugins/Araport/img/navigation_bar.jpg" height="150px">'
                + '</dd>'
                + '<dt>Displaying a Track</dt>'
                + '<dd><ul><li>Turn a track on by checking its box.</li>'
                + '        <li>Turn a track off by selecting the "X" button on the track header.</li>'
                + '    </ul>'
	        + '<img src="plugins/Araport/img/activate_track.jpg" height="250px">'
	        + '<ul>'
	        + '<li>A drop down menu is available for each track.</li>'
	        + '</ul>'
	        + '<img src="plugins/Araport/img/track_submenu.jpg" height="230px">'
                + '</dd>'
                + '<h2 id="search">Search</h2>'
                + '<dd><ul>'
                + '    <li>Jump to a feature by typing its name in the location box and pressing Enter. You can also search by the gene symbol.</li>'
                + '    <li>Jump to a specific region by typing the coordinates into the location box as: <span class="example">ref:start..end</span></li>'
                + '    <li>The metadata that is available in the Faceted list selector is search/filter enabled as well.</li>'
                + '    <li>The track list can be filtered using the text box.</li>'
	        + '</ul>'
	        + '<img src="plugins/Araport/img/filter_by_featureId.jpg" height="150px">'
                + '</dd>'
                + '<dd>'
                + '    <dl class="searchexample">'
		+ '        <dt>Example Searches</dt>'
                + '        <dt>uc0031k.2</dt><dd>searches for the feature named <span class="example">uc0031k.2</span>.</dd>'
                + '        <dt>chr4</dt><dd>jumps to chromosome 4</dd>'
                + '        <dt>chr4:79,500,000..80,000,000</dt><dd>jumps the region on chromosome 4 between 79.5Mb and 80Mb.</dd>'
                + '        <dt>5678</dt><dd>centers the display at base 5,678 on the current sequence</dd>'
                + '    </dl>'
                + '</dd>'
                + '<h2 id="data">Data</h2>'
	        + '<dd>'
	        + '<table class="trackListTable">'
	        + '<tr class="odd"><td class="trackListHeader">Reference Sequence</td><td class="TrackListValue">TAIR10 reference genome sequence</td></tr>'
	    + '<tr class="even"><td class="trackListHeader">Locus</td><td class="TrackListValue">The total transcribed region of an annotated gene including coding and non-coding genes. The arrow indicates the direction of transcription.</td></tr>'
	    + '<tr class="odd"><td class="trackListHeader">Protein Coding Gene Models</td><td class="TrackListValue">Shows the exon (boxes) and intron (lines) structure of each protein-coding transcript with direction of transcription indicated. The gene model confidence rankings are embedded in this track. Documentation for the ranking system can be found at: <ftp: ftp.arabidopsis.org="" home="" tair="" genes="" tair10_genome_release="" tair10_gene_confidence_ranking="" documentation_tair_gene_confidence.pdf=""></ftp:></td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Natural Transposons</td><td class="TrackListValue">Incorporates TE annotation from TAIR and external sources including TIGR and Colot group.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">Unconfirmed Exons</td><td class="TrackListValue">Shows the exons in TAIR10 which are neither supported by experimental evidence nor bioinformatic analysis. The detailed scoring strategy is available at: <ftp: ftp.arabidopsis.org="" home="" tair="" genes="" tair10_genome_release="" tair10_gene_confidence_ranking="" documentation_tair_gene_confidence.pdf=""></ftp:></td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Pseudogenes</td><td class="TrackListValue">Shows the pseudogene transcript.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">RNA-Seq Based Spliced Junctions</td><td class="TrackListValue">RNA-seq data from Ecker and Mockler labs were aligned to TAIR7 using Tophat and Supersplat. The numbers indicate the supporting read counts for each junction.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Natural Transposons (Buisine et al. 2008)</td><td class="TrackListValue">Shows the predicted transposable elements using TAIR7 genome and additional reference sets composed mainly of mosaics of existing genomic sequences.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">ncRNA</td><td class="TrackListValue">Shows the non-coding RNA transcripts.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Tiling Path</td><td class="TrackListValue">Shows the whole-genome annotation including BAC and other genomic clones.</td></tr>'
	    + '<tr class="odd"><td class="trackListHeader">Assembly Gap</td><td class="TrackListValue">Shows the regions where greater than four Ns are included in the current golden path sequence.</td></tr>'
	    + '<tr class="even"><td class="trackListHeader">Tandem Repeat</td><td class="TrackListValue">Regions of tandem repeats including  microsatellite, minisatellite, and satellite, as computed using Tandem Repeats Finder v4.04.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">Assembly Updates</td><td class="TrackListValue">Changes to sequences following the TIGR5 genome release.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">T-DNAs/Transposons</td><td class="TrackListValue">Shows approximate positions of T-DNA and transposon insertion sites.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">Polymorphisms</td><td class="TrackListValue">Shows the positions of polymorphisms , including SNPs, TILLing mutations, small insertions and deletions.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Markers</td><td class="TrackListValue">Shows the positions of genetic markers.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">Microarray ATH1-121501 probes</td><td class="TrackListValue">This genome array is based on the initial release of Arabidopsis genome in 2000 and subsequent TIGR reannotation.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">AtProteome (Baerenfaller et al. 2008)</td><td class="TrackListValue">AtProteome assembles proteomic data of and provides information about identified proteotypic peptides, abundances and tissue and organelle-specific appearance.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">AtPeptide (Castellana et al. 2008)</td><td class="TrackListValue">Proteomics data. Discovery and revision of Arabidopsis genes by proteogenomics Natalie E. Castellana, Samuel H. Payne, Zhouxin Shen, Mario Stanke, Vineet Bafna, and Steven P. Briggs. The white/blue box colour indicates pValue (higher to lower).</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Alternative Splicing Variants (Marquez et al 2012)</td><td class="TrackListValue">A normalized cDNA library was generated from pooled seedling and floral tissues and sequenced with paired-end Illumina.The reads were aligned to the TAIR9 reference genome using Bowtie.TopHat was used to detect the splice junctions. Transcriptome survey reveals increased complexity of the alternative splicing landscape.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">All PEAT tag clusters in whole root (Morton et al. 2014)</td><td class="TrackListValue">A large-scale transcription start site (TSS) data set was generated from 10-day-old wild-type root using paired-end analysis of transcription start site (PEAT) protocol, providing TSS locations based on TAIR10 gene annotation. Each PEAT tag cluster (or “peak”) requires a minimum of 10 reads.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">PEAT coverage in root (Morton et al. 2014)</td><td class="TrackListValue">This track shows the read coverage of raw BAM files and reveals the landscape of transcription initiation in Arabidopsis whole root samples.</td></tr>'
	    + '<tr class="odd"><td class="trackListHeader">Direct RNA Sequencing of wild-type seedling (fwd) (Duc et al. 2013)</td><td class="TrackListValue">This experiment examined the relative 3&#39;UTR positioning and expression in 14-day-old Col-0 seedling samples using Helicos BioSciences Direct RNA Sequencing. This track displays the forward-strand read-depth data from the summed wild-type replicates for this experiment. Each peak indicates the number of reads that end at a given position.</td></tr>'
		        + '<tr class="even"><td class="trackListHeader">Direct RNA Sequencing of wild-type seedling (rev) (Duc et al. 2013)</td><td class="TrackListValue">This experiment examined the relative 3&#39;UTR positioning and expression in 14-day-old Col-0 seedling samples using Helicos BioSciences Direct RNA Sequencing. This track displays the forward-strand read-depth data from the summed wild-type replicates for this experiment. Each peak indicates the number of reads that end at a given position.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">EST and Full-Length cDNA (est2genome)</td><td class="TrackListValue">Arabidopsis thaliana ESTs and FL-cDNAs downloaded from GenBank and mapped to the genome using MAKER optimized est2genome. Data provided courtesy of Yandell Lab.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">Non-Arabidopsis plant protein (SwissProt 201112)</td><td class="TrackListValue">Non-Arabidopsis Plant protein sequences from SwissProt plant (release 2011_12) were mapped to the genome using MAKER optimized protein2genome. Data provided courtesy of Yandell Lab.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">Non-Arabidopsis plant protein (SwissProt 201503)</td><td class="TrackListValue">Protein sequences from 2123 non-Arabidopsis plant species were downloaded from UniprotKB (SwissProt, 20150304) and mapped to TAIR10 genome using MAKER-P protein2genome pipeline.</td></tr>'
	        + '<tr class="even"><td class="trackListHeader">1001 Genomes (Ensembl v.78)</td><td class="TrackListValue">Variation features from Affy_250k, Perlegen_1M, WTCHG and 1001 Genomes, with Ensembl identifiers&nbsp;<img src="https://www.jcvi.org/cms/fileadmin/template/main/images/icons/icon-new.jpg"></td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">SALK TDNA-seq (O&#39;Malley R et al. release Dec 2014)</td><td class="trackListHeader">SALK TDNA-seq (O&#39;Malley R et al. release Dec 2014)&nbsp;<img src="https://www.jcvi.org/cms/fileadmin/template/main/images/icons/icon-new.jpg"></td></tr>'
	        + '<tr class="even"><td class="trackListHeader">RNA-binding protein protected sites (Gosai et al. 2015)</td><td class="TrackListValue">RNA-binding protein protected sites (PPS) were detected using protein interaction profile sequencing (PIP-seq) on total nuclei from 10-day-old seedlings. A total of 61,632 PPSs were detected from two-replicate nuclear PIP-seq experiments, 64.7% of which overlap. Consolidation of two replicates yields 40,131 distinct sites which are presented in this track.</td></tr>'
	        + '<tr class="odd"><td class="trackListHeader">EPIC-CoGe tracks</td><td class="TrackListValue">epigenetic Col-0 mutants (See Faceted List for metadata)</td></tr>'

                + '</table>'
                + '</dd>'
	        + '<h2 id="tracklists">Track Lists</h2>'
		+ ' <dt>Hierarchical Track List</dt>'
                + '<dd>'
                + '<ul>'
                + ' <li>The hierarchical track list displays tracks arranged into various categories and sub-categories. </li>'
                + '</ul>'
	        + '<img src="plugins/Araport/img/track_lists.jpg" height="350px">'
                + '</dd>'
	        + '<dt>Faceted Track Displayer</dt>'
                + '<dd>'
	        + '<ul>'
	        + ' <li>In addition to the tracks available within the hierarchical menu, additional tracks are available through the faceted track displayer. </li>'
	        + ' <li>The comprehensive list of metadata are available on the left hand side and is searchable. The data displayed within the columns are only a subset.</li>'
	        + '<li>The dual display of the hierarchical and faceted list is a plugin we have developed within the Araport project.This enables the capability to partition tracks, placing metadata rich tracks in the “Faceted” selector while using the default “Hierarchical” selector for everything else.</li>'
                + '</ul>'
	        + '<img src="plugins/Araport/img/faceted_list.jpg" height="400px">'
                + '</dd>'
	        + '<h2 id="newfeatures">New Features</h2>'
                + '<dt>SeqLighter v1.0</dt>'
                + '<dd><ul>'
	        + '<li>SeqLighter is a dynamic sequence viewer plugin implemented with the <a target="_blank" href="http://biojs.net/">BioJS framework</a>.</li>'
	        + '   <li>To launch the sequence viewer, right click((Ctrl+click on Mac) a gene model on the ‘Protein Coding Gene Models’ track. Select View Sequence on the drop down menu.</li><br>'
	        + '<img src="plugins/Araport/img/activate_SeqLighter.jpg" height="180px">'
	        + '   <li>This will launch a popup screen displaying the nucleotide sequence of the selected gene model. The default format is in CODATA and there are other sequence formats such as FASTA, PRIDE, and RAW to choose from.</li><br>'
	        + '<img src="plugins/Araport/img/SeqLighter_Intro.jpg" height="450px">'
	        + '<li>Flanking Sequence: An option to add upstream and downstream sequences of the gene model is available in 500bp, 1K, 2K, 3K and 4K. </li><br>'
	        + '<img src="plugins/Araport/img/flanking_region.jpg" height="250px"><br style="clear:both">'
	        + '<br style="clear:both"><li>When one of these are selected, the additional sequence is highlighted in grey.Note: The flanking sequences will not available in the downloaded image.</li>'
	        + '<li>The downloadable formats include PNG, JPEG and SVG.</li>'
	        + '<li>Note: The downloaded images will only be in CODATA format.</li>'
	        + '<li>Highlighting options for exons, introns, utrs and start/stop codon is available. In addition, reverse complement of the selected sequence [including flanking regions] is available. </li><br>'
	        + '<img src="plugins/Araport/img/Seqlighter_highlights.jpg" height="450px"><br>'
	        + '<br><li>When reverse complement is enabled, other highlighting options will not be available. The highlighting is available for all of the sequence formats such as FASTA, PRIDE, CODATA and RAW. </li>'
                + '</ul>'
                + '</dd>'
                + '<dt>Variant Tracks Features</dt>'
                + '<dd><ul>'
                + '<li>Select the 1001 Genomes track from the track list.</li>'
	        + '<img src="plugins/Araport/img/VariantTrack_open.jpg" height="300px"><br>'
                + '<br><li>The 1001 Genomes variant track is color coded by the most severe variant consequence using <a target="_blank" href="http://plants.ensembl.org/info/genome/variation/predicted_data.html#consequence_type_table">Ensembl&#39;s consequence diagram</a>.</li>'
	        + '<li>The color legend is available on the track drop down menu.</li><br>'
	        + '<img src="plugins/Araport/img/Ensembl_ColorLegend.jpg" height="450px"><br>'
	        + '<br><li>The track can be filtered by the consequence variation type (missense, intron_variant, frameshift variants, etc):</li><br>'
	        + '<img src="plugins/Araport/img/track_filters.jpg" height="320px"><br>'
	        + '<br><li>The track can also be filtered by the variant type (deletion, insertion, snv, etc.):</li>'
	        + '<img src="plugins/Araport/img/track_filter2.jpg" height="300px"><br>'
                + '<br><li>To view the variant feature dialogue box, right click(Ctrl+click on Mac) a variant marker on the 1001 Genomes track and select "View Variation Feature from the menu.</li>'
	        + '<li>The variant feature dialogue box retrieves the data using Ensembl&#39;s <a target="_blank" href="http://rest.ensemblgenomes.org/">REST API</a>.</li><br>'
	        + '<img src="plugins/Araport/img/Wait_for_results.jpg" height="300px"><br>'
	        + '<br><li>The variation features popup displays genotype, population and phenotype data for the selected feature.</li><br>'
	        + '<img src="plugins/Araport/img/Example_ResultPage.jpg" height="700px"><br />'
                + '</ul></dd>'
                + '<h2 id="FutureDevelopments">Future Developments</h2>'
                + '<dd><ul><li>Feature list export to interact with Thalemine, BLAST and other Science Apps</li>'
                + '        <li>Add filters for various tracks</li>'
                + '        <li>Enable web services</li>'
                + '    </ul>'
                + '</dd>'
                + '</dl>'
                + '</div>'

            ;
    }
});
});
